kernel/term.mli

```diff
@@ -127,13 +127,27 @@ val decompose_prod_n : int -> constr -> (Name.t Context.binder_annot * constr) l
    Raise a user error if not enough lambdas. *)
 val decompose_lam_n : int -> constr -> (Name.t Context.binder_annot * constr) list * constr
 
+(** Simultaneous decompose_lam_n / decompose_prod_n *)
+val decompose_lam_prod_n : int
```

Comment:
Not sure it's really worth having this function, why not just call the 2 in the comment?

Reply:
That's a possibility; this PR is really a draft [I should maybe have submitted as such] in the sense that I'm far from sure on what do we want from shrinking.

Only point I'm sure is that "shrinking" does belong as core `Declare` API.